core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -165,6 +165,20 @@ public void testMultipleRleWithDifferentValue(Type type)
         testAppend(type, input);
     }
 
+    @Test(dataProvider = "types")
+    public void testMultipleRleWithTheSameValueProduceRle(Type type)
+    {
+        PositionsAppender positionsAppender = POSITIONS_APPENDER_FACTORY.create(type, null, 10);
+
+        Block value = notNullBlock(type, 1);
+        positionsAppender.append(allPositions(3), rleBlock(value, 3));
+        positionsAppender.append(allPositions(2), rleBlock(value, 2));
+
+        Block actual = positionsAppender.build();
+        assertEquals(actual.getPositionCount(), 5);
+        assertTrue(actual instanceof RunLengthEncodedBlock, actual.toString());
+    }
+
```

Comment:
I believe many code paths in `AdaptivePositionAppender` has not been tested here. Like switching to flat. Please analyze coverage and add tests

Reply:
I don't think we use anything fancy. Line converage is far from perfect but better than nothing.
I can't run it myself at the moment but if you say its coevered then it's ok for me